test/e2e/page-objects/pages/onboarding/onboarding-privacy-settings-page.ts

```diff
@@ -0,0 +1,191 @@
+    await this.navigateToGeneralSettings();
+    console.log('Adding custom network');
+    await this.driver.clickElement(this.addCustomNetworkButton);
+    await this.driver.waitForSelector(this.addCustomNetworkFormMessage);
+    await this.driver.fill(this.networkNameInput, networkName);
+    await this.driver.fill(this.chainIdInput, chainId.toString());
+    await this.driver.fill(this.currencySymbolInput, currencySymbol);
+    // Add rpc url
+    await this.driver.clickElement(this.addRpcUrlDropDown);
+    await this.driver.clickElement(this.addRpcUrlButton);
+    await this.driver.waitForSelector(this.addRpcUrlDialogMessage);
+    await this.driver.fill(this.rpcUrlInput, networkUrl);
+    await this.driver.clickElement(this.confirmAddRpcUrlButton);
+    await this.driver.clickElementAndWaitToDisappear(
+      this.confirmAddCustomNetworkButton,
+    );
+    // Navigate back to default privacy settings
+    await this.driver.clickElement(this.categoryBackButton);
+  }
+
+  /**
+   * Navigate back to the onboarding complete page.
+   */
+  async navigateBackToOnboardingCompletePage(): Promise<void> {
+    console.log('Navigate back to onboarding complete page');
+    // Wait until the onboarding carousel has stopped moving otherwise the click has no effect.
+    await this.driver.waitForElementToStopMoving(
+      this.privacySettingsBackButton,
+    );
+    await this.driver.clickElementAndWaitToDisappear(
+      this.privacySettingsBackButton,
+    );
+  }
+
+  async navigateToGeneralSettings(): Promise<void> {
+    console.log('Navigate to general settings');
+    await this.check_pageIsLoaded();
+    await this.driver.clickElement(this.generalSettings);
+    await this.driver.waitForSelector(this.generalSettingsMessage);
+  }
+
+  /**
+   * Go to assets settings and toggle options, then navigate back.
+   */
+  async toggleAssetsSettings(): Promise<void> {
+    console.log('Toggle advanced assets settings in privacy settings');
+    await this.check_pageIsLoaded();
+    await this.driver.clickElement(this.assetsSettings);
+    await this.driver.waitForSelector(this.assetsSettingsMessage);
+    await Promise.all(
+      (
+        await this.driver.findClickableElements(this.assetsPrivacyToggle)
+      ).map((toggle) => toggle.click()),
+    );
+    await this.driver.clickElement(this.categoryBackButton);
+  }
+
+  /**
+   * Go to general settings and toggle options, then navigate back.
+   */
+  async toggleBasicFunctionalitySettings(): Promise<void> {
+    console.log('Toggle basic functionality settings in privacy settings');
+    await this.navigateToGeneralSettings();
+    await this.driver.clickElement(this.basicFunctionalityToggle);
+    await this.driver.waitForSelector(this.basicFunctionalityTurnOffMessage);
+    await this.driver.clickElement(this.basicFunctionalityCheckbox);
+    await this.driver.clickElement(this.basicFunctionalityTurnOffButton);
+    await this.driver.clickElement(this.categoryBackButton);
```

Comment:
we cannot wait for the element to disappear as it will remain in the DOM, but we can wait for the element to stop moving to, before proceeding with next steps.

```suggestion
    await this.driver.clickElement(this.categoryBackButton);
        await this.driver.waitForElementToStopMoving(this.categoryBackButton);
```

https://github.com/user-attachments/assets/1715fd32-df9b-4b17-a34b-505fb46c398f

Reply:
Thanks for the suggestion, i have added it in commit https://github.com/MetaMask/metamask-extension/pull/28036/commits/c80c16f25c65f611a657d5b8b82b6395182f30b3